internal/p2p/channel.go

```diff
@@ -15,15 +15,7 @@ type Envelope struct {
 	To        types.NodeID  // receiver (empty if inbound)
 	Broadcast bool          // send to all connected peers (ignores To)
 	Message   proto.Message // message payload
-
-	// channelID is for internal Router use, set on outbound messages to inform
-	// the sendPeer() goroutine which transport channel to use.
-	//
-	// FIXME: If we migrate the Transport API to a byte-oriented multi-stream
-	// API, this will no longer be necessary since each channel will be mapped
-	// onto a stream during channel/peer setup. See:
-	// https://github.com/tendermint/spec/pull/227
-	channelID ChannelID
+	ChannelID ChannelID
```

Comment:
If we change to a callback approach as @creachadair suggests, or similar, I believe this will no longer need to be a public member of Envelope since we would no longer need to `switch` over it.

Reply:
So I think we can make this private as we don't end up using it outside of the blocksync reactor, but to be honest, having it on incoming messages and accessible allows us to process messages with some amount of fanout (rather than serialize processing all messages?) and so that doesn't really hurt us?

I think in the libp2p world, It'd be nice to be able to register message types but have each reactor only have to manage a single incoming stream of messages, so that reactor implementors have a much more constrained concurrency problem.